reload: wait for pending tasks to submit and pause the workflow (#5592)
cylc/cylc-flow: CHANGES.md

### Enhancements

[#5992](https://github.com/cylc/cylc-flow/pull/5992) -

> The scheduler will now wait for preparing tasks to submit before attempting
> to perform a reload and will also pause/unpause the workflow.

Reviewer: Suggestion: I think that `when you do x => then the scheduler will do y` is more readable.

```suggestion
Before trying to reload the workflow definition, the scheduler will
wait for preparing tasks to submit, and pause the workflow. 
After reload the scheduler will unpause the workflow.
```
